File: Google-Hashcode-2016/src/google/hashcode/pkg2016/GridItem.java
package google.hashcode.pkg2016;

public class GridItem {
    public int x;
    public int y;
    
    public GridItem(int x, int y) {
        this.x = x;
        this.y = y;
    }
    
    public int distanceBetween(GridItem other) {
        double rowDiff = this.x - other.x;
        double colDiff = this.y - other.y;
        double sum = Math.pow(rowDiff, 2) + Math.pow(colDiff, 2);
        return (int)Math.ceil(Math.sqrt(sum));
    }
}

File: GridItem.cs
using System;

namespace HashCode2016 {
    public class GridItem {

        public int X { get; set; }
        public int Y { get; set; }

        public GridItem(int x, int y) {
            X = x;
            Y = y;
        }

        public int DistanceBetween(GridItem other) {
            double sum = Math.Pow(X - other.X, 2) + Math.Pow(Y - other.Y, 2);
            return (int)Math.Ceiling(Math.Sqrt(sum));
        }
    }
}
